staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go

```diff
@@ -51,18 +52,39 @@ func (az *Cloud) getStorageAccounts(matchingAccountType, matchingAccountKind, re
 	for _, acct := range result {
 		if acct.Name != nil && acct.Location != nil && acct.Sku != nil {
 			storageType := string((*acct.Sku).Name)
-			if matchingAccountType != "" && !strings.EqualFold(matchingAccountType, storageType) {
+			if accountOptions.Type != "" && !strings.EqualFold(accountOptions.Type, storageType) {
 				continue
 			}
 
-			if matchingAccountKind != "" && !strings.EqualFold(matchingAccountKind, string(acct.Kind)) {
+			if accountOptions.Kind != "" && !strings.EqualFold(accountOptions.Kind, string(acct.Kind)) {
 				continue
 			}
 
 			location := *acct.Location
-			if matchingLocation != "" && !strings.EqualFold(matchingLocation, location) {
+			if accountOptions.Location != "" && !strings.EqualFold(accountOptions.Location, location) {
 				continue
 			}
+
+			if len(accountOptions.VirtualNetworkResourceIDs) > 0 {
+				if acct.AccountProperties == nil || acct.AccountProperties.NetworkRuleSet == nil ||
+					acct.AccountProperties.NetworkRuleSet.VirtualNetworkRules == nil {
+					continue
+				}
+
+				found := false
+				for _, subnetID := range accountOptions.VirtualNetworkResourceIDs {
+					for _, rule := range *acct.AccountProperties.NetworkRuleSet.VirtualNetworkRules {
+						if strings.EqualFold(*rule.VirtualNetworkResourceID, subnetID) && rule.Action == storage.Allow {
```

Comment:
nit: check nil of `rule.VirtualNetworkResourceID`, or use `to.String(rule.VirtualNetworkResourceID)` replace `*rule.VirtualNetworkResourceID`

Reply:
good catch, fixed.